tests_manual/syscall_module_path/README.txt

```diff
@@ -0,0 +1,52 @@
+Title: module load PATH null records
+
+
+Description:
+This test detects kernels that flood the audit logs with hundreds of
+NULL PATH SYSCALL auxiliary records and the modules that provoke them.
+
+It does this by setting up an audit rule in the config file to log all
+init_module and finit_module syscalls, then once the machine is rebooted
+the test script detects SYSCALL records with NULL PATH records along
+with their provoking modules since the most recent boot.
+
+
+How to run the test:
+
+1) Change to the working directory of the test.
+
+	cd tests_manual/syscall_module_path
+
+2) Set up the test, which adds the tests_manual-syscall_module_path
+   audit rule file to the /etc/audit/rules.d/ audit rule directory to
+   log all init_module and finit_module syscalls.
+
+	./setup
+
+3) Reboot the machine to trigger the test.
+
+	shutdown -r now
+
+4) Run the test itself that looks for *init_module syscall records since
+   this last boot and searches for any NULL PATH records listed in the
+   syscall event.
+
+	cd tests_manual/syscall_module_path
+	./test
+
+   If there are, the test will fail, and if ATS_DEBUG is set, it will
+   list which modules cause the records.
```

Comment:
# ./test
    1..3
    # Running under perl version 5.026001 for linux
    # Current time local: Mon Oct 23 17:49:03 2017
    # Current time GMT:   Mon Oct 23 21:49:03 2017
    # Using Test.pm version 1.30
    not ok 1
    # Test 1 got: "0" (./test at line 113)
    #   Expected: "1"
    #  ./test line 113 is: ok( $found_syscall, 1 ); # Were module    syscall records found?
    not ok 2
    # Test 2 got: "0" (./test at line 114)
    #   Expected: "1"
    #  ./test line 114 is: ok( $found_spam,    1 ); # Were module     tracefs or debugfs PATH records detected?
    ok 3
    # ATS_DEBUG=1 ./test
    1..3
    # Running under perl version 5.026001 for linux
    # Current time local: Mon Oct 23 17:49:11 2017
    # Current time GMT:   Mon Oct 23 21:49:11 2017
    # Using Test.pm version 1.30
    Checking for module null PATH messages since boottime:     2017:10:23 17:1:24
    not ok 1
    # Test 1 got: "0" (./test at line 113)
    #   Expected: "1"
    #  ./test line 113 is: ok( $found_syscall, 1 ); # Were module syscall records found?
    not ok 2
    # Test 2 got: "0" (./test at line 114)
    #   Expected: "1"
    #  ./test line 114 is: ok( $found_spam,    1 ); # Were module tracefs or debugfs PATH records detected?
    ok 3